src/components/20-molecules/image-upload/index.js

```diff
@@ -0,0 +1,433 @@
+const AttachFileIcon = svg([AttachFileSvg]);
+const DeleteForeverIcon = svg([DeleteForeverSvg]);
+const ClearIcon = svg([ClearSvg]);
+const ImageUploadGroupIcon = svg([ImageUploadGroupSvg]);
+
+const ACCEPTED_FILE_TYPES = 'image/jpg, image/jpeg, application/pdf, image/png';
+
+/* helperfunctions */
+export const getBytesFromKilobyte = kilobyte => 1024 * kilobyte;
+
+class AXAImageUpload extends LitElement {
+  static get tagName() {
+    return 'axa-image-upload';
+  }
+
+  static get styles() {
+    return css`
+      ${unsafeCSS(styles)}
+    `;
+  }
+
+  static get properties() {
+    return {
+      inputFileText: { type: String },
+      maxSizeOfSingleFileKB: { type: Number },
+      maxSizeOfAllFilesKB: { type: Number },
+      maxNumberOfFiles: { type: Number },
+      showImageOverview: { type: Boolean },
+      icon: { type: String },
+      deleteStatusText: { type: String },
+      addStatusText: { type: String },
+      fileTooBigStatusText: { type: String },
+      filesTooBigStatusText: { type: String },
+      tooManyFilesStatusText: { type: String },
+      orText: { type: String },
+      infoText: { type: String },
+    };
+  }
+
+  constructor() {
+    super();
+    this.inputFileText = 'Upload file';
+    this.maxSizeOfSingleFileKB = 100;
+    this.maxSizeOfAllFilesKB = 500;
+    this.maxNumberOfFiles = 10;
+    this.showImageOverview = false;
+    this.icon = 'cloud-upload';
+    this.deleteStatusText = 'Delete';
+    this.addStatusText = 'Add more';
+    this.fileTooBigStatusText = `File size exceeds maximum size`;
+    this.filesTooBigStatusText = `File sizes exceed maximum size`;
+    this.tooManyFilesStatusText = `You exceeded the maximum number of files`;
+    this.orText = 'or';
+    this.infoText = 'Drag and drop to upload your file';
+    this.files = [];
+    this.faultyFiles = [];
+    this.allFiles = [];
+
+    this.sizeOfAllFilesByte = 0;
+    this.allDroppedFiles = 0;
+    this.isFileMaxReached = false;
+
+    this.globalErrorMessage = '';
+    this.addMoreInputFile = '';
+    this.showAddMoreInputFile = '';
+  }
+
+  handleAddMoreInputClick() {
+    /* trigger inputFile component after clicking the "fake" inputFile (addMoreInput) */
+    this.inputFile.querySelector('input').click();
+  }
+
+  handleImageUploadButtonChange(e) {
+    const { files } = e.target;
+    this.addFiles(files);
+  }
+
+  handleImageUploadDropZoneDragover(e) {
+    /* prevent default browser behavior to execute the link that comes with the event */
+    e.preventDefault();
+    if (!this.isFileMaxReached) {
+      e.dataTransfer.dropEffect = 'copy';
+      this.dropZone.classList.add('m-image-upload__dropzone_dragover');
+    }
+  }
+
+  handleImageUploadDropZoneDragleave() {
+    this.dropZone.classList.remove('m-image-upload__dropzone_dragover');
+  }
+
+  handleImageUploadDropZoneDrop(e) {
+    /* prevent browser to display the file fullscreen */
+    e.preventDefault();
+
+    if (this.isFileMaxReached) {
+      return;
```

Comment:
It does in `this.handleMaxNumberOfFiles()` because the variable is set after. But after the max is reached and the user wants to drop even more files, the images won't be compressed unnecessarily.